### PR TITLE
Sass 862

### PR DESCRIPTION
**Pull request by joeyymoore**
### Description
Creation of 'amount of tips and others not on P60' page

Jira Link
https://jira.tools.tax.service.gov.uk/browse/SASS-862

##### Before Reviewing
- [ ]  Have you pulled the branch down?
- [ ]  Have you assigned yourself to the PR?
- [ ]  Have you moved the task to “in review” on JIRA?
- [ ]  Have you checked to ensure all dependencies are up to date?
- [ ]  Have you checked to ensure its been rebased against the current version of master?

##### Whilst Reviewing
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [ ]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current master that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of master?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/views/employment/AmountOfTipsOnP60View.scala.html [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            )
+
+
+
+        @button()
+    }
+}
+
+@contentHtml = {
+    @if(amount != None){
```
Comment (Dula-J) [inline app/views/employment/AmountOfTipsOnP60View.scala.html:65]: Option has a method `.isDefined` you could use here

File: it/controllers/employment/AmountOfTipsOnP60ControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+          buttonCheck(expectedButtonText, continueButtonSelector)
+          formPostLinkCheck(continueLink, continueButtonFormSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+      }
+    }
+
+    "redirect to the 'did you receive any payments that are not on your p60' page when they've previously answered no" which {
+      lazy val result: WSResponse = {
+        authoriseAgentOrIndividual(false)
```
Comment (Dula-J) [inline it/controllers/employment/AmountOfTipsOnP60ControllerISpec.scala:185]: this should be user.isAgent

File: it/controllers/employment/AmountOfTipsOnP60ControllerISpec.scala [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+        dropEmploymentDB()
+        insertCyaData(empDataWithNoToP60Question, User(mtditid,None,nino,sessionId,"Individual")(fakeRequest))
+        urlGet(amountOfTipsOnP60PageUrl, welsh = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(2021)))
```
Comment (Dula-J) [inline it/controllers/employment/AmountOfTipsOnP60ControllerISpec.scala:188]: welsh = user.isWelsh

File: it/controllers/employment/AmountOfTipsOnP60ControllerISpec.scala [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+        result.status shouldBe OK
+      }
+
+      titleCheck("Did you receive any payments that are not on your P60?")
+      h1Check("Did you receive any payments that are not on your P60?")
+    }
+
+    "redirect to the 'did you receive any payments that are not on your p60' page when they've not answered the previous question" which {
+      lazy val result: WSResponse = {
+        authoriseAgentOrIndividual(false)
```
Comment (Dula-J) [inline it/controllers/employment/AmountOfTipsOnP60ControllerISpec.scala:203]: same as above

File: it/controllers/employment/AmountOfTipsOnP60ControllerISpec.scala [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+        dropEmploymentDB()
+        insertCyaData(empDataWithUnansweredP60Question, User(mtditid,None,nino,sessionId,"Individual")(fakeRequest))
+        urlGet(amountOfTipsOnP60PageUrl, welsh = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(2021)))
```
Comment (Dula-J) [inline it/controllers/employment/AmountOfTipsOnP60ControllerISpec.scala:206]: same as above

File: app/controllers/employment/AmountOfTipsOnP60Controller.scala [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.EmploymentSessionService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import utils.SessionHelper
+import views.html.employment.{AmountOfTipsOnP60View, CheckEmploymentDetailsView}
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class AmountOfTipsOnP60Controller @Inject()(implicit val cc: MessagesControllerComponents,
```
Comment (Dula-J) [inline app/controllers/employment/AmountOfTipsOnP60Controller.scala:35]: should the controller name be tip payments that are not on P60? or maybe EmploymentOtherPaymentsController?

**Review by Dula-J — CHANGES_REQUESTED** [review]
See comments

File: app/views/employment/AmountOfTipsOnP60View.scala.html [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+    @formWithCSRF(action = controllers.employment.routes.AmountOfTipsOnP60Controller.submit(taxYear, employmentId)) {
+
+
+            @inputText(
+                form,
+                id = "amount",
+                name = "amount",
+                heading = headingForUse,
+                headingClasses = "govuk-! -margin-bottom-2",
```
Comment (BenCosfordHMRC) [inline app/views/employment/AmountOfTipsOnP60View.scala.html:50]: "govuk-! -margin-bottom-2" space should be removed

File: app/views/employment/AmountOfTipsOnP60View.scala.html (hunk 1 above)
Comment (joeyymoore) [reply app/views/employment/AmountOfTipsOnP60View.scala.html]: updated

File: it/controllers/employment/AmountOfTipsOnP60ControllerISpec.scala (hunk 2 above)
Comment (joeyymoore) [reply it/controllers/employment/AmountOfTipsOnP60ControllerISpec.scala]: updated

File: it/controllers/employment/AmountOfTipsOnP60ControllerISpec.scala (hunk 3 above)
Comment (joeyymoore) [reply it/controllers/employment/AmountOfTipsOnP60ControllerISpec.scala]: updated

File: it/controllers/employment/AmountOfTipsOnP60ControllerISpec.scala (hunk 4 above)
Comment (joeyymoore) [reply it/controllers/employment/AmountOfTipsOnP60ControllerISpec.scala]: updated

File: it/controllers/employment/AmountOfTipsOnP60ControllerISpec.scala (hunk 5 above)
Comment (joeyymoore) [reply it/controllers/employment/AmountOfTipsOnP60ControllerISpec.scala]: updated

File: app/controllers/employment/AmountOfTipsOnP60Controller.scala (hunk 6 above)
Comment (joeyymoore) [reply app/controllers/employment/AmountOfTipsOnP60Controller.scala]: updated to be OtherPaymentsAmount view/controller

File: app/views/employment/AmountOfTipsOnP60View.scala.html (hunk 7 above)
Comment (joeyymoore) [reply app/views/employment/AmountOfTipsOnP60View.scala.html]: updated

**Pull request merged**